### DISCUSSION TITLE
Can't build flutter in Android Studio when use module 'UnleashProxyClientSwift'

### DISCUSSION THREAD

**Discussion by vynguyenbao28** [Unleash/unleash#6059] [Q&A]
I got this error when building flutter on android studio mac m1, still able to build on xcode
Detail: Swift Compiler Error (Xcode): Could not find module 'UnleashProxyClientSwift' for target 'x86_64-apple-ios-simulator'; found: arm64-apple-ios-simulator, at: /Users/vy.nb/Downloads/CODE/kvfnb-app-man/build/ios/Debug-dev-iphonesimulator/UnleashProxyClientSwift.swiftmodule
/Users/vy.nb/Downloads/CODE/kvfnb-app-man/ios/Runner/Unleash/UnleashConfiguration.swift:9:7

**Comment by nunogois** [comment]
It looks like there's a mismatch between the architectures being targeted during your build process. Specifically, your build is targeting the `x86_64` architecture for the iOS simulator, but the available 'UnleashProxyClientSwift' module is compiled for the `arm64` architecture, which is the native architecture for M1 Macs.

Are you using Rosetta for any part of the process? That may cause these types of mismatches.

Here are some suggestions:
 - Ensure that both Xcode and Flutter are up-to-date.
 - Check the project's build settings in Xcode. Does the ARCHS setting include `arm64` for the iOS Simulator? Make sure `arm64` is not part of "excluded architectures".
 - When running `pod install`, make sure it installs packages that target `arm64`.
 
 If you're using CocoaPods, adding this post-install script in your Podfile may help ensure that all pods are built for `arm64`:
 
 ```rb
 post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ONLY_ACTIVE_ARCH'] = 'NO'
      config.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = ''
    end
  end
end
```
 
 Let us know if any of this helps.

**Reply by vynguyenbao28** [reply]
I am forced to use Rosetta mode. Is this the main cause?

**Reply by andreas-unleash** [reply]
It could be a number of places.  Let me confirm: are you using the latest version of Android studio? (Some older versions didn't play well with Rosetta)

**Comment by nunogois** [comment]
This feature isn't currently on our roadmap. If you're a paying customer, submitting a support ticket helps us prioritize future updates based on user needs.
